### PR TITLE
pcapplusplus: bump version to v23.09

### PR DESCRIPTION
**Pull request by seladb**
Specify library name and version:  **pcapplusplus/23.09**

This PR updates the PcapPlusPlus recipe to the latest version v23.09.
This version includes a major change in the build system - moving from a custom build system to CMake.
I asked in the Slack channel what's the best way to approach this change, and I was told it's better to create a new folder and a new `conanfile.py`, which is what this PR contains.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by seladb** [comment]
@ericLemanissier I'm not sure why but the CI is in pending state for a long time. I tried to close and open the PR, but it didn't work

**Comment by seladb** [comment]
@ericLemanissier I'm not sure how to fix the Conan v2 pipeline CI error. When I run it locally indeed I can't find any packages for `npcap`:

```bash
seladb$ conan list "npcap/1.70:*" -r=conancenter    
conancenter
  npcap
    npcap/1.70
      revisions
        92438da8500f8940a33d2bb159fe5de4 (2022-09-15 19:45:46 UTC)
          packages
```

https://conan.io/center/recipes/npcap?version=1.70

Maybe you can create the packages for `npcap` and upload it to ConanCenter?

**Comment by SpaceIm** [comment]
I'm not a conan team member, I can't build missing binaries of npcap.

**Comment by seladb** [comment]
> I'm not a conan team member, I can't build missing binaries of npcap.

Thank you @SpaceIm ! I opened a PR to update Npcap to Conan 2.0: https://github.com/conan-io/conan-center-index/pull/20399
Hopefully, this should help to unblock this PR

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [inline recipes/pcapplusplus/cmake/conanfile.py:60]:
```suggestion
        self.cpp_info.libs = ["Pcap++", "Packet++", "Common++"]
        if self.settings.os == "Macos":
            self.cpp_info.frameworks = ["CoreFoundation", "SystemConfiguration"]
        elif self.settings.os == "Windows":
           self.settings.system_libs = ["ws2_32", "iphlpapi"]
```

I see some system libraries are required to build pcap++:

https://github.com/seladb/PcapPlusPlus/blob/v23.09/Pcap%2B%2B/CMakeLists.txt#L69

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [inline recipes/pcapplusplus/cmake/conanfile.py:12]: Is correct to affirm that pcap++ only supports static library? I see shared option option available in the CMake file: https://github.com/seladb/PcapPlusPlus/blob/v23.09/CMakeLists.txt#L95


Plus, I can build in my machine shared library without errors:

```
$ cmake -S . -B build -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/tmp/install -DCMAKE_VERBOSE_MAKEFILE=ON -DPCAPPP_BUILD_TESTS=OFF -DPCAPPP_BUILD_EXAMPLES=OFF -DCMAKE_TOOLCHAIN_FILE=build/Release/generators/conan_toolchain.cmake
...
[ 27%] Linking CXX shared library libCommon++.so
cd /home/conan/project/build/Common++ && /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/Common++.dir/link.txt --verbose=1
/usr/bin/g++ -fPIC -m64 -O3 -DNDEBUG -m64 -shared -Wl,-soname,libCommon++.so.23.09 -o libCommon++.so.23.09 CMakeFiles/Common++.dir/src/GeneralUtils.cpp.o CMakeFiles/Common++.dir/src/IpAddress.cpp.o CMakeFiles/Common++.dir/src/IpUtils.cpp.o CMakeFiles/Common++.dir/src/Logger.cpp.o CMakeFiles/Common++.dir/src/MacAddress.cpp.o CMakeFiles/Common++.dir/src/OUILookup.cpp.o CMakeFiles/Common++.dir/src/PcapPlusPlusVersion.cpp.o CMakeFiles/Common++.dir/src/SystemUtils.cpp.o CMakeFiles/Common++.dir/src/TablePrinter.cpp.o 
cd /home/conan/project/build/Common++ && /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_symlink_library libCommon++.so.23.09 libCommon++.so.23.09 libCommon++.so
```

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [inline recipes/pcapplusplus/cmake/conanfile.py:20]:
```suggestion
        "immediate_mode": [True, False],
        "fPIC": [True, False],
```
I see no rule in Cmake file forcing or forbidden fpic 🤔

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [inline recipes/pcapplusplus/cmake/conanfile.py:23]:
```suggestion
        "immediate_mode": False,
        "fPIC": True,
```

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [inline recipes/pcapplusplus/cmake/conanfile.py:24]:
```suggestion
    }

    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

```
Manage fPIC in case building on Windows.

**Review by uilianries — COMMENTED** [review]
@seladb Thank you for your contribution, I have some suggestion and a question related to shared option.

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [inline recipes/pcapplusplus/cmake/conanfile.py:59]:
```suggestion
    def package_info(self):
        self.cpp_info.set_property("cmake_file_name", "PcapPlusPlus")
        self.cpp_info.set_property("cmake_target_name", "PcapPlusPlus::PcapPlusPlus")
```

I see the upstream generated cmake files are nor lower case:

```
Install the project...
/opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /tmp/install/lib/pkgconfig/PcapPlusPlus.pc
-- Installing: /tmp/install/lib/cmake/pcapplusplus/FindPCAP.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusTargets.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusTargets-release.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusConfig.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusConfigVersion.cmake
```


Plus, I found individual targets, like ` PcapPlusPlus::Packet++ PcapPlusPlus::Pcap++ PcapPlusPlus::Common++` that much better. It's possible to add here in cpp_info too as [property](https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#properties)

File: recipes/pcapplusplus/cmake/test_package/CMakeLists.txt
Comment (uilianries) [inline recipes/pcapplusplus/cmake/test_package/CMakeLists.txt:4]:
```suggestion
find_package(PcapPlusPlus REQUIRED CONFIG)
```

File: recipes/pcapplusplus/cmake/test_package/CMakeLists.txt
Comment (uilianries) [inline recipes/pcapplusplus/cmake/test_package/CMakeLists.txt:7]:
```suggestion
target_link_libraries(${PROJECT_NAME} PcapPlusPlus::PcapPlusPlus)
```

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (SpaceIm) [inline recipes/pcapplusplus/cmake/conanfile.py:39]:
```suggestion
        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (SpaceIm) [inline recipes/pcapplusplus/cmake/conanfile.py:8]:
```suggestion
required_conan_version = ">=1.52.0"
```

1.52.0 is sufficient (even less maybe, but there are some tedious issues before 1.52.0, so I think this version is fine)

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (SpaceIm) [inline recipes/pcapplusplus/cmake/conanfile.py:45]: No CMakeDeps? How dependencies are discovered?

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (SpaceIm) [reply recipes/pcapplusplus/cmake/conanfile.py:59]: There is also this `PcapPlusPlus.pc`, so `pkg_config_name` property should be set accordingly.

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:12]: Shared libraries are not well supported on all platforms, so I prefer to keep the static option only

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:45]: PcapPlusPlus doesn't use `find_package()` because it doesn't have dependencies that must be compiled with the library. Hence `CMakeDeps` is not needed

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:20]: I don't think `fPIC` is relevant for building static libraries, am I wrong? 🤔

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:23]: See this comment: https://github.com/conan-io/conan-center-index/pull/20296#discussion_r1356052062

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:24]: See this comment: https://github.com/conan-io/conan-center-index/pull/20296#discussion_r1356052062

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py]: Fixed in these commits: https://github.com/conan-io/conan-center-index/pull/20296/commits/fe96dbda54ca1e70a1960957a445a4f6c080dfc3, https://github.com/conan-io/conan-center-index/pull/20296/commits/5d7919d56df970d0f4c56e9864452df1ac50b512

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:25]: Fixed in these commits: https://github.com/conan-io/conan-center-index/pull/20296/commits/fe96dbda54ca1e70a1960957a445a4f6c080dfc3, https://github.com/conan-io/conan-center-index/pull/20296/commits/5d7919d56df970d0f4c56e9864452df1ac50b512

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:63]: Fixed in this commit: https://github.com/conan-io/conan-center-index/pull/20296/commits/6bc67b3f751a5c9c6830adf4ea4275de25575d72

I didn't add Linux and FreeBSD because PcapPlusPlus requires C++11 so pthreads and libm should be part of the standard library

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py]: Fixed in https://github.com/conan-io/conan-center-index/pull/20296/commits/fe96dbda54ca1e70a1960957a445a4f6c080dfc3

File: recipes/pcapplusplus/cmake/test_package/CMakeLists.txt
Comment (seladb) [reply recipes/pcapplusplus/cmake/test_package/CMakeLists.txt]: Fixed in https://github.com/conan-io/conan-center-index/pull/20296/commits/9f304cbee9a2a61149277c00d741bb16ec521467

File: recipes/pcapplusplus/cmake/test_package/CMakeLists.txt
Comment (seladb) [reply recipes/pcapplusplus/cmake/test_package/CMakeLists.txt]: Fixed in https://github.com/conan-io/conan-center-index/pull/20296/commits/9f304cbee9a2a61149277c00d741bb16ec521467

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:62]: Fixed in Fixed in https://github.com/conan-io/conan-center-index/pull/20296/commits/9f304cbee9a2a61149277c00d741bb16ec521467

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py]: Fixed in https://github.com/conan-io/conan-center-index/pull/20296/commits/79b408e995cb18fff78ea0132d6a07fc705b14e3

**Comment by seladb** [comment]
@uilianries @SpaceIm thank you for the review! I addressed your comments, can you please re-review?

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [reply recipes/pcapplusplus/cmake/conanfile.py:20]: We manage fPIC always for static libraries in CCI. For shared we remove because fPIC should enabled always.

Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:20]: Got it, updated here: https://github.com/conan-io/conan-center-index/pull/20296/commits/667883dd2315215e75910859d6ed6024f8ed63e5

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:24]: Updated here: https://github.com/conan-io/conan-center-index/pull/20296/commits/667883dd2315215e75910859d6ed6024f8ed63e5

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:26]: Updated here: https://github.com/conan-io/conan-center-index/pull/20296/commits/667883dd2315215e75910859d6ed6024f8ed63e5

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [reply recipes/pcapplusplus/cmake/conanfile.py:49]: @seladb but what about the libpcap listed as requirement?

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (uilianries) [inline recipes/pcapplusplus/cmake/conanfile.py:71]:
```suggestion
            self.cpp_info.system_libs = ["ws2_32", "iphlpapi"]
```
Bad indentation.

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py:49]: @uilianries `libpcap`/`WinPcap`/`Npcap` aren't compliled as part of PcapPlusPlus build system. It's assumed the compliled binaries already exist on the system:
- On macOS it's easy because libpcap comes as part of the OS
- On Linux a user can usually download and install libpcap using `apt` or `yum`
- On Windows the user should download [WinPcap developer's pack](https://www.winpcap.org/devel.htm) OR [Npcap SDK](https://nmap.org/npcap/guide/npcap-devguide.html)

You can read more about it here: https://pcapplusplus.github.io/docs/install

File: recipes/pcapplusplus/cmake/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/cmake/conanfile.py]: Fix in https://github.com/conan-io/conan-center-index/pull/20296/commits/443139b9c3df298afd4dd759209f12c855548bcb

**Review by uilianries — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Comment by seladb** [comment]
Thank you @uilianries and @danimtb for reviewing this PR!

@SpaceIm I think you're the last reviewer. I'd really appreciate it if you could review so we can finish and merge this PR

**Comment by uilianries** [comment]
@seladb We actually need 2 approvals only, one from Conan team and another from community, or 2 from Conan team. It's documented https://github.com/conan-io/conan-center-index/blob/master/docs/review_process.md#rule-of-2-reviews 

SpaceIm review is really welcome, but the CI will ignore the third count and merge early.

**Comment by seladb** [comment]
> @seladb We actually need 2 approvals only, one from Conan team and another from community, or 2 from Conan team. It's documented https://github.com/conan-io/conan-center-index/blob/master/docs/review_process.md#rule-of-2-reviews
> 
> SpaceIm review is really welcome, but the CI will ignore the third count and merge early.

Thank you @uilianries for this info. I really appreciate your help, as well as @SpaceIm 's and @danimtb 's help in reviewing and merging this PR 🙏

**Pull request merged**
